src/data/finset/basic.lean

```diff
@@ -526,6 +526,15 @@ protected theorem induction {α : Type*} {p : finset α → Prop} [decidable_eq
     { rw [insert_val, ndinsert_of_not_mem m] }
   end) nd
 
+protected theorem induction' {α : Type*} {p : finset α → Prop}
```

Comment:
We already have an `induction_on'`, I think it's confusing having these mean different things. Perhaps this should be called `induction''` or `cons_induction`?

It would be good to add an `on` variant too, since those are often easier to use in term-mode proofs.